From ff299bbed74e9603ea7c780de8805110d88dec28 Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Wed, 01 Dec 2010 11:37:59 +0000
Subject: Replace GDK macro usage with proper function name

---
diff --git a/libgnome-desktop/gnome-bg-crossfade.c b/libgnome-desktop/gnome-bg-crossfade.c
index df6a285..02a120e 100644
--- a/libgnome-desktop/gnome-bg-crossfade.c
+++ b/libgnome-desktop/gnome-bg-crossfade.c
@@ -373,7 +373,7 @@ animations_are_disabled (GnomeBGCrossfade *fade)
 static void
 draw_background (GnomeBGCrossfade *fade)
 {
-	if (GDK_WINDOW_TYPE (fade->priv->window) == GDK_WINDOW_ROOT) {
+	if (gdk_window_get_window_type (fade->priv->window) == GDK_WINDOW_ROOT) {
                 XClearArea (GDK_WINDOW_XDISPLAY (fade->priv->window),
                             GDK_WINDOW_XID (fade->priv->window),
                             0, 0,
@@ -494,7 +494,7 @@ gnome_bg_crossfade_start (GnomeBGCrossfade *fade,
 	g_return_if_fail (fade->priv->fading_surface != NULL);
 	g_return_if_fail (fade->priv->end_surface != NULL);
 	g_return_if_fail (!gnome_bg_crossfade_is_started (fade));
-	g_return_if_fail (GDK_WINDOW_TYPE (window) != GDK_WINDOW_FOREIGN);
+	g_return_if_fail (gdk_window_get_window_type (window) != GDK_WINDOW_FOREIGN);
 
 	source = g_timeout_source_new (1000 / 60.0);
 	g_source_set_callback (source,
--
cgit v0.8.3.1
