--- totem-pl-parser-2.23.2/totem-plparser-uninstalled.pc.in	2008-07-05 17:29:40.000000000 -0700
+++ totem-pl-parser-2.23.2/totem-plparser-uninstalled.pc.in	2008-07-05 17:29:50.000000000 -0700
@@ -9,6 +9,6 @@ Name: totem-plparser
 Description: Totem Playlist Parser library
 Version: @VERSION@
 Requires: gtk+-2.0
-Requires.private: libxml-2.0 camel-1.2
+Requires.private: libxml-2.0
 Libs: ${pc_top_builddir}/${pcfiledir}/plparse/libtotem-plparser.la
 Cflags: -I${pc_top_builddir}/${pcfiledir}/plparse
--- totem-pl-parser-2.23.2/configure.in	2008-07-05 17:32:07.000000000 -0700
+++ totem-pl-parser-2.23.2/configure.in	2008-07-05 17:32:15.000000000 -0700
@@ -76,7 +76,7 @@ fi
 AM_CONDITIONAL(HAVE_HAL, test x"$with_hal" = xyes)
 
 dnl Check for packages for building libtotem-plparser.la
-PKG_CHECK_MODULES(TOTEM_PLPARSER, [glib-2.0 >= $GLIB_REQS gtk+-2.0 libxml-2.0 camel-1.2 gio-2.0])
+PKG_CHECK_MODULES(TOTEM_PLPARSER, [glib-2.0 >= $GLIB_REQS gtk+-2.0 libxml-2.0 gio-2.0])
 
 PKG_CHECK_MODULES([TOTEM_PLPARSER_MINI], [gio-2.0])
 
@@ -80,11 +80,11 @@
 
 PKG_CHECK_MODULES([TOTEM_PLPARSER_MINI], [gio-2.0])
 
-GNOME_COMMON_INIT
-GNOME_DEBUG_CHECK
-GNOME_COMPILE_WARNINGS([maximum])
-GNOME_CXX_WARNINGS
-GNOME_MAINTAINER_MODE_DEFINES
+dnl GNOME_COMMON_INIT
+dnl GNOME_DEBUG_CHECK
+dnl GNOME_COMPILE_WARNINGS([maximum])
+dnl GNOME_CXX_WARNINGS
+dnl GNOME_MAINTAINER_MODE_DEFINES
 
 if test "$GCC" = "yes" ; then
 	CFLAGS="\
@@ -102,6 +102,8 @@
 AC_SUBST([AM_LDFLAGS])
 
-GTK_DOC_CHECK(1.0)
+dnl GTK_DOC_CHECK(1.0)
+AM_CONDITIONAL([GTK_DOC_USE_LIBTOOL], [0])
+AM_CONDITIONAL([ENABLE_GTK_DOC], [0])
 
 AC_OUTPUT([
 Makefile
--- totem-pl-parser-2.23.2/totem-plparser.pc.in	2008-07-05 17:29:17.000000000 -0700
+++ totem-pl-parser-2.23.2/totem-plparser.pc.in	2008-07-05 17:29:27.000000000 -0700
@@ -9,6 +9,6 @@ Name: totem-plparser
 Description: Totem Playlist Parser library
 Version: @VERSION@
 Requires: gtk+-2.0
-Requires.private: libxml-2.0 camel-1.2
+Requires.private: libxml-2.0
 Libs: -L${libdir} -ltotem-plparser
 Cflags: -I${includedir}/totem-pl-parser/1/plparser
--- totem-pl-parser-2.23.2/plparse/totem-pl-parser.c	2008-07-05 17:30:11.000000000 -0700
+++ totem-pl-parser-2.23.2/plparse/totem-pl-parser.c	2008-07-05 17:31:27.000000000 -0700
@@ -103,7 +103,6 @@
 #ifndef TOTEM_PL_PARSER_MINI
 #include <gobject/gvaluecollector.h>
 #include <gtk/gtk.h>
-#include <camel/camel-mime-utils.h>
 
 #include "totem-pl-parser.h"
 #include "totemplparser-marshal.h"
@@ -1733,7 +1732,7 @@ totem_pl_parser_parse_date (const char *
 	D(g_message ("Failed to parse duration '%s' using the ISO8601 parser", date_str));
 
 	/* Fall back to RFC 2822 date parsing */
-	return camel_header_decode_date (date_str, NULL);
+	return 0;
 }
 
 #endif /* !TOTEM_PL_PARSER_MINI */
