Author: compnerd@compnerd.org
Upstream: no
Reason: fix compile failure
--- totem-pl-parser-2.23.2/configure.in	2008-07-13 17:45:20.000000000 -0700
+++ totem-pl-parser-2.23.2/configure.in	2008-07-13 17:46:28.000000000 -0700
@@ -76,7 +76,7 @@
 AM_CONDITIONAL(HAVE_HAL, test x"$with_hal" = xyes)
 
 dnl Check for packages for building libtotem-plparser.la
-PKG_CHECK_MODULES(TOTEM_PLPARSER, [glib-2.0 >= $GLIB_REQS gtk+-2.0 libxml-2.0 gio-2.0])
+PKG_CHECK_MODULES(TOTEM_PLPARSER, [glib-2.0 >= $GLIB_REQS gthread-2.0 gtk+-2.0 libxml-2.0 gio-2.0])
 
 PKG_CHECK_MODULES([TOTEM_PLPARSER_MINI], [gio-2.0])
 
