--- gtkmm-2.13.7/configure.in	2008-09-01 12:03:54.000000000 -0700
+++ gtkmm-2.13.7/configure.in	2008-09-01 12:04:10.000000000 -0700
@@ -292,11 +292,11 @@
 
 # Add an --enable-use-deprecations configure option:
 AC_ARG_ENABLE(deprecations,
-              [AC_HELP_STRING([--enable-use-deprecations],
+              [AC_HELP_STRING([--enable-deprecations],
                               [warn about deprecated usages [default=no]])],,
               [enable_deprecations=no])
               
-if test "x$enable_use_deprecations" = "xyes"; then
+if test "x$enable_deprecations" = "xyes"; then
    DISABLE_DEPRECATED_CFLAGS="\
 -DG_DISABLE_DEPRECATED \
 -DGDK_DISABLE_DEPRECATED \
