From fc4c54353e176cebaac92eb9480a4fb63e2be270 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Fri, 5 Mar 2010 10:48:47 +0000
Subject: [PATCH] add support for explicitly enabling/disabling bluetooth

---
 configure.ac |   16 +++++++++-------
 1 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index a928b6a..5205049 100644
--- a/configure.ac
+++ b/configure.ac
@@ -101,13 +101,15 @@ fi
 AC_SUBST(DBUS_SYS_DIR)
 
 dnl Check for gnome-bluetooth
-PKG_CHECK_MODULES(GNOME_BLUETOOTH,
-		  gconf-2.0
-		  gnome-bluetooth-1.0 >= 2.27.6
-		  libnm-util >= 0.8
-		  libnm-glib >= 0.8,
-		  have_gbt=yes, have_gbt=no)
-AM_CONDITIONAL(HAVE_GBT, test x"$have_gbt" = "xyes")
+AC_ARG_ENABLE([bluetooth],
+	      AS_HELP_STRING([--disable-bluetooth], [disable support for bluetooth DUNs]),,
+	      [enable_bluetooth=yes])
+
+if test x"$enable_bluetooth" = x"yes" ; then
+    PKG_CHECK_MODULES(GNOME_BLUETOOTH, gconf-2.0 gnome-bluetooth-1.0 >= 2.27.6 libnm-util >= 0.8 libnm-glib >= 0.8)
+fi
+
+AM_CONDITIONAL(HAVE_GBT, test x"$enable_bluetooth" = x"yes")
 
 GLIB_GENMARSHAL=`pkg-config --variable=glib_genmarshal glib-2.0`
 AC_SUBST(GLIB_GENMARSHAL)
-- 
1.7.0.1

