From c1008ee0e31b9ada5ebf4854988687394281693b Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Wed, 22 Dec 2010 17:00:56 +0000
Subject: Don't use gdk_spawn to launch gdmflexiserver

---
diff --git a/gnome-session/gsm-manager.c b/gnome-session/gsm-manager.c
index c215fed..39288ea 100644
--- a/gnome-session/gsm-manager.c
+++ b/gnome-session/gsm-manager.c
@@ -926,8 +926,9 @@ cancel_end_session (GsmManager *manager)
 static void
 manager_switch_user (GsmManager *manager)
 {
+        GAppInfo *app;
+        GAppLaunchContext *context;
         GError  *error;
-        gboolean res;
         char    *command;
 
         command = g_strdup_printf ("%s %s",
@@ -935,13 +936,16 @@ manager_switch_user (GsmManager *manager)
                                    GDM_FLEXISERVER_ARGS);
 
         error = NULL;
-        res = gdk_spawn_command_line_on_screen (gdk_screen_get_default (),
-                                                command,
-                                                &error);
+        context = (GAppLaunchContext*)gdk_app_launch_context_new ();
+        app = g_app_info_create_from_commandline (command, "gdmflexiserver", 0, &error);
+        if (app)
+                g_app_info_launch (app, NULL, context, &error);
 
         g_free (command);
+        g_object_unref (context);
+        g_object_unref (app);
 
-        if (! res) {
+        if (! error) {
                 g_debug ("GsmManager: Unable to start GDM greeter: %s", error->message);
                 g_error_free (error);
         }
--
cgit v0.8.3.1
