From ce7828bbd9446fd08edfaca9815b76c1314b99ed Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Thu, 02 Dec 2010 23:05:48 +0000
Subject: Fix build with current GTK+

---
diff --git a/gnome-session/test-inhibit.c b/gnome-session/test-inhibit.c
index 68cc0a7..8a08ab9 100644
--- a/gnome-session/test-inhibit.c
+++ b/gnome-session/test-inhibit.c
@@ -87,7 +87,7 @@ do_inhibit_for_window (GdkWindow *window)
         app_id = "nautilus";
         reason = "A file transfer is in progress.";
 #endif
-        toplevel_xid = GDK_DRAWABLE_XID (window);
+        toplevel_xid = gdk_x11_window_get_xid (window);
         flags = GSM_INHIBITOR_FLAG_LOGOUT
                 | GSM_INHIBITOR_FLAG_SWITCH_USER
                 | GSM_INHIBITOR_FLAG_SUSPEND;
--
cgit v0.8.3.1
