From c4133bacb43e4ca65b98fb1a6e8397ae2befa12d Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Tue, 6 Jul 2010 22:20:17 -0700
Subject: [PATCH 2/2] remove usage of sealed fields

---
 src/backend/bacon-resize.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/backend/bacon-resize.c b/src/backend/bacon-resize.c
index f934a15..d067aac 100644
--- a/src/backend/bacon-resize.c
+++ b/src/backend/bacon-resize.c
@@ -227,7 +227,7 @@ bacon_resize_resize (BaconResize *resize)
 	g_return_if_fail (GTK_IS_WIDGET (resize->priv->video_widget));
 	g_return_if_fail (gtk_widget_get_realized (resize->priv->video_widget));
 
-	xdisplay = GDK_DRAWABLE_XDISPLAY (resize->priv->video_widget->window);
+	xdisplay = GDK_DRAWABLE_XDISPLAY (gtk_widget_get_window (GTK_WIDGET (resize->priv->video_widget)));
 	if (xdisplay == NULL)
 		return;
 
@@ -308,7 +308,7 @@ bacon_resize_restore (BaconResize *resize)
 	if (resize->priv->xr_screen_conf == NULL)
 		return;
 
-	xdisplay = GDK_DRAWABLE_XDISPLAY (resize->priv->video_widget->window);
+	xdisplay = GDK_DRAWABLE_XDISPLAY (gtk_widget_get_window (GTK_WIDGET (resize->priv->video_widget)));
 	if (xdisplay == NULL)
 		return;
 
-- 
1.7.1

