Author: Saleem Abdulrasool <compnerd@compnerd.org>
Reason: Fix tests
Upstream: Bug #554969
================================================================================
--- glib-2.19.2/tests/slice-threadinit.c	2008-12-01 20:34:25.000000000 -0800
+++ glib-2.19.2/tests/slice-threadinit.c	2008-12-01 20:34:40.000000000 -0800
@@ -23,7 +23,7 @@
 #define N_PAGES                 (101)                   /* number of pages to sample */
 #define SAMPLE_SIZE             (7)
 #define PAGE_SIZE               (128)                   /* must be <= minimum GSlice alignment block */
-#define MAGAZINE_PROBES         { 77, 265, 347 }        /* block sizes hopefully unused by g_thread_init */
+#define MAGAZINE_PROBES         { 101, 265, 347 }       /* block sizes hopefully unused by g_thread_init */
 #define MAX_PROBE_TRIALS        (1031)                  /* must be >= maximum magazine size */
 
 #define ALIGN(size, base)       ((base) * (gsize) (((size) + (base) - 1) / (base)))
