From 9058a2384d62d31c378734de100ccf3db14d6049 Mon Sep 17 00:00:00 2001
From: Brett Witherspoon <spoonb@exherbo.org>
Date: Thu, 13 Jan 2011 21:03:57 -0600
Subject: [PATCH] gnome-keyring-optionalize-libcap

---
 configure.in |   24 ++++++++++++++++--------
 1 files changed, 16 insertions(+), 8 deletions(-)

diff --git a/configure.in b/configure.in
index 13a9018..755a3e8 100644
--- a/configure.in
+++ b/configure.in
@@ -450,16 +450,24 @@ fi
 # libcap2
 #
 
-AC_CHECK_LIB([cap], [cap_get_proc], have_libcap="yes", have_libcap="no")
-
-if test $have_libcap = yes; then
-   AC_DEFINE(HAVE_LIBCAP, 1, [Have libcap2 package, libcap library])
-   DAEMON_LIBS="$DAEMON_LIBS -lcap"
-else
-   AC_MSG_WARN([libcap2 (or development headers) is not installed])
-fi
+AC_ARG_WITH(libcap,
+            AC_HELP_STRING([--without-libcap],
+            [Build without libcap and disable linux capabilities]),,
+            [with_libcap=yes])
+
+libcap_status="no"
+if test $with_libcap != "no"; then
+   AC_CHECK_LIB([cap], [cap_get_proc], have_libcap="yes", have_libcap="no")
+
+   if test $have_libcap = yes; then
+      AC_DEFINE(HAVE_LIBCAP, 1, [Have libcap2 package, libcap library])
+      DAEMON_LIBS="$DAEMON_LIBS -lcap"
+   else
+      AC_MSG_WARN([libcap2 (or development headers) is not installed])
+   fi
 
 libcap_status=$have_libcap
+fi
 
 # --------------------------------------------------------------------
 # Debug mode
-- 
1.7.3.5

