From af2f78cbac6215cb22a771b5ca11b7dfa34de184 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Tue, 10 May 2011 23:19:11 -0700
Subject: [PATCH] optionalise libcap-ng check

---
 configure.ac |   30 ++++++++++++++++++++----------
 1 files changed, 20 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index d9e947a..0e9409c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -451,18 +451,28 @@ fi
 # libcap-ng
 #
 
-AC_CHECK_LIB([cap-ng], [capng_clear], have_libcapng="yes", have_libcapng="no")
-
-if test "$have_libcapng" = "yes"; then
-   AC_DEFINE(HAVE_LIBCAPNG, 1, [Have libcap-ng package, libcap-ng library])
-   DAEMON_LIBS="$DAEMON_LIBS -lcap-ng"
-else
-   have_lipcapng="no"
-   AC_MSG_WARN([libcap-ng (or development headers) is not installed])
+AC_ARG_WITH([libcap-ng],
+            [AC_HELP_STRING([--without-libcap-ng],
+                            [build without libcap-ng (disables Linux capabilities support)])],,
+            [with_libcap_ng=auto])
+
+if test x"$with_libcap_ng" != x"no" ; then
+    AC_CHECK_LIB([cap-ng], [capng_clear],
+                 [
+                   with_libcap_ng="yes"
+                   AC_DEFINE([HAVE_LIBCAPNG], [1], [have libcap-ng headers and library])
+                   DAEMON_LIBS="$DAEMON_LIBS -lcap-ng"
+                 ],
+                 [
+                   with_libcap_ng="no"
+                   if test x"$with_libcap_ng" = x"yes" ; then
+                     AC_MSG_ERROR([libcap-ng support requested, but package not found])
+                   fi
+                 ])
 fi
 
-AM_CONDITIONAL(WITH_CAPS, test "$have_libcapng" = "yes")
-libcapng_status=$have_libcapng
+AM_CONDITIONAL([WITH_CAPS], [test x"$with_libcap_ng" = x"yes"])
+libcapng_status="$with_libcap_ng"
 
 # --------------------------------------------------------------------
 # Debug mode
-- 
1.7.4.3

