--- gnome-keyring-2.25.5/pkcs11/rpc-layer/gck-rpc-dispatch.c	2009-01-19 18:51:19.000000000 -0800
+++ gnome-keyring-2.25.5/pkcs11/rpc-layer/gck-rpc-dispatch.c	2009-01-19 18:51:59.000000000 -0800
@@ -202,7 +202,7 @@
 	GckRpcMessage *msg;
 	const unsigned char *data;
 	unsigned char valid;
-	uint32_t n_data;
+	size_t n_data;
 	
 	assert (cs);
 	
--- gnome-keyring-2.25.5/pkcs11/rpc-layer/gck-rpc-module.c	2009-01-19 18:48:24.000000000 -0800
+++ gnome-keyring-2.25.5/pkcs11/rpc-layer/gck-rpc-module.c	2009-01-19 18:50:57.000000000 -0800
@@ -758,7 +758,7 @@
 	
 	/* If not valid, then just the length is encoded, this can signify CKR_BUFFER_TOO_SMALL */
 	if (!valid) {
-		if (!egg_buffer_get_uint32 (&msg->buffer, msg->parsed, &msg->parsed, &vlen))
+		if (!egg_buffer_get_uint32 (&msg->buffer, msg->parsed, &msg->parsed, (uint32_t *)&vlen))
 			return PARSE_ERROR;
 		
 		*len = vlen;
