From 68c050735a3db53207c8769330eba17a73a7f4a1 Mon Sep 17 00:00:00 2001
From: Brett Witherspoon <spoonb@exherbo.org>
Date: Sun, 27 Mar 2011 18:54:11 -0500
Subject: [PATCH] remove setOrientation method

---
 .../WebCore/platform/graphics/SimpleFontData.cpp   |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/Source/WebCore/platform/graphics/SimpleFontData.cpp b/Source/WebCore/platform/graphics/SimpleFontData.cpp
index 82f770d..5cf8d86 100644
--- a/Source/WebCore/platform/graphics/SimpleFontData.cpp
+++ b/Source/WebCore/platform/graphics/SimpleFontData.cpp
@@ -210,7 +210,6 @@ SimpleFontData* SimpleFontData::verticalRightOrientationFontData() const
         m_derivedFontData = DerivedFontData::create(isCustomFont());
     if (!m_derivedFontData->verticalRightOrientation) {
         FontPlatformData verticalRightPlatformData(m_platformData);
-        verticalRightPlatformData.setOrientation(Horizontal);
         m_derivedFontData->verticalRightOrientation = new SimpleFontData(verticalRightPlatformData, isCustomFont(), false, true);
     }
     return m_derivedFontData->verticalRightOrientation.get();
-- 
1.7.4.1

