Index: trunk/Source/WebCore/platform/graphics/gtk/DrawingBufferGtk.cpp
===================================================================
--- trunk/Source/WebCore/platform/graphics/gtk/DrawingBufferGtk.cpp	(revision 100032)
+++ trunk/Source/WebCore/platform/graphics/gtk/DrawingBufferGtk.cpp	(revision 101067)
@@ -37,6 +37,9 @@
                              const IntSize& size,
                              bool multisampleExtensionSupported,
-                             bool packedDepthStencilExtensionSupported)
-    : m_context(context)
+                             bool packedDepthStencilExtensionSupported,
+                             bool separateBackingTexture)
+    : m_separateBackingTexture(separateBackingTexture)
+    , m_scissorEnabled(false)
+    , m_context(context)
     , m_size(-1, -1)
     , m_multisampleExtensionSupported(multisampleExtensionSupported)
@@ -45,7 +48,12 @@
     , m_colorBuffer(0)
     , m_depthStencilBuffer(0)
+    , m_depthBuffer(0)
+    , m_stencilBuffer(0)
     , m_multisampleFBO(0)
     , m_multisampleColorBuffer(0)
 {
+    // Support for a separate backing texture has only been enabled for
+    // the chromium port.
+    ASSERT(!m_separateBackingTexture);
     ASSERT(m_fbo);
     if (!m_fbo) {
