From c6e8cc385bee387f501041d244070dcd5a00b51a Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Wed, 22 Dec 2010 17:08:04 -0800
Subject: [PATCH] fix for newer gtk+

---
 configure.in |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index 9072baa..ff75491 100644
--- a/configure.in
+++ b/configure.in
@@ -16,7 +16,7 @@ dnl ***************************************************************************
 dnl *** Minimum library versions for libgnomekbd                            ***
 dnl ***************************************************************************
 GLIB_REQUIRED=2.18
-GDK_REQUIRED=2.18
+GDK_REQUIRED=2.91.7
 GTK_REQUIRED=2.90
 LIBXKLAVIER_REQUIRED=5.0
 
@@ -63,7 +63,7 @@ PKG_CHECK_MODULES(GDK, gdk-3.0 >= $GDK_REQUIRED)
 AC_SUBST(GDK_CFLAGS)
 AC_SUBST(GDK_LIBS)
 
-PKG_CHECK_MODULES(GTK, gtk+-3.0 >= $GTK_REQUIRED gdk-x11-3.0 >= $GDK_REQUIRED)
+PKG_CHECK_MODULES(GTK, gtk+-3.0 >= $GTK_REQUIRED)
 
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
-- 
1.7.3.2

