From c98cd071769cabcd3de965c970ebe8a1bc981642 Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Mon, 19 Dec 2011 16:36:13 +0000
Subject: Fix typo in windows-live backend

Signed-off-by: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
---
diff --git a/src/goabackend/goawindowsliveprovider.h b/src/goabackend/goawindowsliveprovider.h
index 8b846b0..2ed9a4d 100644
--- a/src/goabackend/goawindowsliveprovider.h
+++ b/src/goabackend/goawindowsliveprovider.h
@@ -34,7 +34,7 @@
 G_BEGIN_DECLS
 
 #define GOA_TYPE_WINDOWS_LIVE_PROVIDER   (goa_windows_live_provider_get_type ())
-#define GOA_WINDOWS_LIVE_PROVIDER(o)     (G_TYPE_CHECK_INSTANCE_CAST ((o), GOA_TYPE_WINDOWS_LIVE_PROVIDER, GoaFacebookProvider))
+#define GOA_WINDOWS_LIVE_PROVIDER(o)     (G_TYPE_CHECK_INSTANCE_CAST ((o), GOA_TYPE_WINDOWS_LIVE_PROVIDER, GoaWindowsLiveProvider))
 #define GOA_IS_WINDOWS_LIVE_PROVIDER(o)  (G_TYPE_CHECK_INSTANCE_TYPE ((o), GOA_TYPE_WINDOWS_LIVE_PROVIDER))
 
 GType goa_windows_live_provider_get_type (void) G_GNUC_CONST;
--
cgit v0.9.0.2
