From 35c92c41f5f833f7f148ec362a3659aa95e1ef1e Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Sun, 19 Dec 2010 16:07:17 -0800
Subject: [PATCH] make gdict compile against latest gtk+

---
 gnome-dictionary/libgdict/gdict-defbox.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/gnome-dictionary/libgdict/gdict-defbox.c b/gnome-dictionary/libgdict/gdict-defbox.c
index 80415f4..5d2f6a0 100644
--- a/gnome-dictionary/libgdict/gdict-defbox.c
+++ b/gnome-dictionary/libgdict/gdict-defbox.c
@@ -2853,7 +2853,11 @@ gdict_defbox_set_font_name (GdictDefbox *defbox,
   else
     font_desc = NULL;
 
+#if GTK_CHECK_VERSION (2, 91, 6)
+  gtk_widget_override_font (priv->text_view, font_desc);
+#else
   gtk_widget_modify_font (priv->text_view, font_desc);
+#endif
 
   if (font_desc)
     pango_font_description_free (font_desc);
-- 
1.7.3.2

