From 5cbfc48caf76ced2e28ee06c9e40523273601dc6 Mon Sep 17 00:00:00 2001
From: Dan Winship <danw@gnome.org>
Date: Fri, 02 Dec 2011 09:21:44 +0000
Subject: bump version to 2.37.2.1 for the request API changes

---
diff --git a/configure.ac b/configure.ac
index b839ec4..899f7af 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@ dnl *** Initialize automake and set version ***
 dnl *******************************************
 
 AC_PREREQ(2.63)
-AC_INIT([libsoup],[2.37.2],[http://bugzilla.gnome.org/enter_bug.cgi?product=libsoup])
+AC_INIT([libsoup],[2.37.2.1],[http://bugzilla.gnome.org/enter_bug.cgi?product=libsoup])
 AC_CONFIG_SRCDIR([libsoup-2.4.pc.in])
 AC_CONFIG_MACRO_DIR([m4])
 
@@ -16,10 +16,10 @@ SOUP_API_VERSION=2.4
 AC_SUBST(SOUP_API_VERSION)
 
 # Increment on interface addition. Reset on removal.
-SOUP_AGE=4
+SOUP_AGE=5
 
 # Increment on interface add, remove, or change.
-SOUP_CURRENT=5
+SOUP_CURRENT=6
 
 # Increment on source change. Reset when CURRENT changes.
 SOUP_REVISION=0
--
cgit v0.9.0.2
