From f4a7b2d97e354674738f2a43424efd03a306a5f7 Mon Sep 17 00:00:00 2001
From: Brett Witherspoon <spoonb@exherbo.org>
Date: Mon, 28 Feb 2011 00:20:55 +0000
Subject: [PATCH] test: don't create a dbus-launch babysitter process

---
 test/run-with-dbus |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/test/run-with-dbus b/test/run-with-dbus
index deae1e6..da889e0 100755
--- a/test/run-with-dbus
+++ b/test/run-with-dbus
@@ -116,7 +116,7 @@ if ! test -z "$START_SESSION_BUS" ; then
 
     echo "Running dbus-launch --config-file=$CONFIG_FILE" >&2
 
-    eval `dbus-launch --exit-with-session --sh-syntax --config-file=$CONFIG_FILE 2>"$STDERR_LOGFILE"`
+    eval `dbus-launch --sh-syntax --config-file=$CONFIG_FILE 2>"$STDERR_LOGFILE"`
 
     if test -z "$DBUS_SESSION_BUS_PID" ; then
         die "Failed to launch message bus for test script to run"
-- 
1.7.4.1

