From bea9f53efc9f8339985be3c86b93b1b7458c75de Mon Sep 17 00:00:00 2001
From: Brett Witherspoon <spoonb@exherbo.org>
Date: Sat, 5 Feb 2011 20:05:51 -0600
Subject: [PATCH] add explicit option for cups

---
 configure.ac |   18 ++++++++++++------
 1 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 39b0b9b..8a4996d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -108,12 +108,18 @@ PKG_CHECK_MODULES(PULSEAUDIO,
 AC_SUBST(PULSEAUDIO_CFLAGS)
 AC_SUBST(PULSEAUDIO_LIBS)
 
-AC_CHECK_HEADERS([cups/cups.h cups/http.h cups/ipp.h], have_cups=yes, have_cups=no)
-if test x$have_cups = xyes ; then
-  CUPS_LIBS=-lcups
-  AC_SUBST(CUPS_LIBS)
-else
-  AC_MSG_WARN(*** Printers panel will not be built (CUPS header files not found) ***)
+AC_ARG_WITH([cups],
+            AS_HELP_STRING([--with-cups], [enable CUPS printer management support]),,
+            with_cups=yes)
+
+if test x"$with_cups" != x"no" ; then
+    AC_CHECK_HEADERS([cups/cups.h cups/http.h cups/ipp.h], have_cups=yes, have_cups=no)
+    if test x$have_cups = xyes ; then
+        CUPS_LIBS=-lcups
+        AC_SUBST(CUPS_LIBS)
+    else
+        AC_MSG_WARN(*** Printers panel will not be built (CUPS header files not found) ***)
+    fi
 fi
 AM_CONDITIONAL(BUILD_PRINTERS, [test x$have_cups = xyes])
 
-- 
1.7.4

