From da3d7a8faeb17d39aa8d9db28a079d2e030a24ae Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Sun, 19 Dec 2010 19:06:15 -0800
Subject: [PATCH] add explicit option for cheese

---
 configure.ac |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7393bad..e995968 100644
--- a/configure.ac
+++ b/configure.ac
@@ -116,10 +116,16 @@ AC_SUBST(GTK_ENGINE_DIR)
 PKG_CHECK_MODULES(GLIB, glib-2.0)
 
 PKG_CHECK_MODULES(POLKIT, polkit-gobject-1 >= 0.97)
-PKG_CHECK_MODULES(CHEESE, gstreamer-0.10 cheese-gtk >= 2.29.90, have_cheese=yes, have_cheese=no)
 
-if test x$have_cheese = xyes ; then
-        AC_DEFINE(HAVE_CHEESE, 1, [Define to 1 to enable cheese webcam support])
+AC_ARG_WITH([cheese],
+            AS_HELP_STRING([--with-cheese], [enable cheese webcam support]),,
+            with_cheese=yes)
+
+if test x"$with_cheese" != x"no" ; then
+   PKG_CHECK_MODULES(CHEESE, gstreamer-0.10 cheese-gtk >= 2.29.90, have_cheese=yes, have_cheese=no)
+   if test x$have_cheese = xyes ; then
+      AC_DEFINE(HAVE_CHEESE, 1, [Define to 1 to enable cheese webcam support])
+   fi
 fi
 
 AC_DEFINE_UNQUOTED([ISO_CODES_PREFIX],["`$PKG_CONFIG --variable=prefix iso-codes`"],[ISO codes prefix])
-- 
1.7.3.2

