--- gobject-introspection-0.6.6/girepository/ginvoke.c	2009-12-12 14:10:25.000000000 -0800
+++ gobject-introspection-0.6.6/girepository/ginvoke.c	2009-12-12 14:15:08.000000000 -0800
@@ -247,6 +247,7 @@
   if (ffi_prep_cif (&cif, FFI_DEFAULT_ABI, n_invoke_args, rtype, atypes) != FFI_OK)
     goto out;
 
+  g_return_val_if_fail (return_value, FALSE);
   ffi_call (&cif, func, return_value, args);
 
   if (local_error)
@@ -435,6 +436,7 @@
   if (ffi_prep_cif (&cif, FFI_DEFAULT_ABI, n_args, rtype, atypes) != FFI_OK)
     return;
 
+  g_return_val_if_fail (rvalue, FALSE);
   ffi_call (&cif, marshal_data ? marshal_data : cc->callback, rvalue, args);
 
   if (return_gvalue && G_VALUE_TYPE (return_gvalue))
