From f666dbb42b97823154723ef3b720f5b449ed5484 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Wed, 22 Dec 2010 19:06:51 -0800
Subject: [PATCH] remove usage of gdk_spawn_command_line_on_screen

---
 src/eog-window.c |   12 +++---------
 1 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/src/eog-window.c b/src/eog-window.c
index 1eb9e3c..1da7cfc 100644
--- a/src/eog-window.c
+++ b/src/eog-window.c
@@ -2640,15 +2640,9 @@ eog_window_cmd_show_hide_bar (GtkAction *action, gpointer user_data)
 static void
 wallpaper_info_bar_response (GtkInfoBar *bar, gint response, EogWindow *window)
 {
-	if (response == GTK_RESPONSE_YES) {
-		GdkScreen *screen;
-
-		screen = gtk_widget_get_screen (GTK_WIDGET (window));
-		gdk_spawn_command_line_on_screen (screen,
-						  "gnome-appearance-properties"
-						  " --show-page=background",
-						  NULL);
-	}
+	if (response == GTK_RESPONSE_YES)
+		g_spawn_command_line_async ("gnome-appearance-properties --show-page=background",
+					    NULL);
 
 	/* Close message area on every response */
 	eog_window_set_message_area (window, NULL);
-- 
1.7.3.2

