From 6cb2ce2c72ce91f4f56d9110d7326e5ede86d7f9 Mon Sep 17 00:00:00 2001
From: Felix Riemann <friemann@gnome.org>
Date: Sat, 12 Jun 2010 16:48:54 +0000
Subject: Make forgotten GSettings keys use the key macros

---
diff --git a/src/eog-window.c b/src/eog-window.c
index 0dd6ab4..6289584 100644
--- a/src/eog-window.c
+++ b/src/eog-window.c
@@ -424,10 +424,11 @@ eog_window_collection_mode_changed_cb (GSettings *settings,
 
 	priv = EOG_WINDOW (user_data)->priv;
 
-	position = g_settings_get_int (settings, "image_collection_position");
+	position = g_settings_get_int (settings,
+				       EOG_CONF_UI_IMAGE_COLLECTION_POSITION);
 
 	resizable = g_settings_get_boolean (settings,
-					    "image_collection_resizable");
+					    EOG_CONF_UI_IMAGE_COLLECTION_RESIZABLE);
 
 	if (priv->collection_position == position &&
 	    priv->collection_resizable == resizable)
--
cgit v0.8.3.1
