From 0b3ed281583e99d4b219600636bea31868d8f180 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Thu, 4 Mar 2010 13:03:43 +0000
Subject: [PATCH] only check for libpanel if building the applets

---
 configure.ac |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 26fc294..a8f864b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -167,12 +167,6 @@ PKG_CHECK_MODULES(GNOME, [
 AC_SUBST(GNOME_CFLAGS)
 AC_SUBST(GNOME_LIBS)
 
-PKG_CHECK_MODULES(PANEL, [
- libpanelapplet-2.0 >= $LIBPANEL_REQUIRED])
-AC_SUBST(PANEL_CFLAGS)
-AC_SUBST(PANEL_LIBS)
-
-
 PKG_CHECK_MODULES(GDK, [
  gdk-2.0 >= $GDK_REQUIRED
  gdk-x11-2.0 >= $GDK_REQUIRED])
@@ -304,6 +298,11 @@ AC_MSG_CHECKING([whether to build applet support])
 have_applets=no
 if test x$enable_applets = xyes ; then
     have_applets=yes
+
+    PKG_CHECK_MODULES(PANEL, [libpanelapplet-2.0 >= $LIBPANEL_REQUIRED])
+    AC_SUBST(PANEL_CFLAGS)
+    AC_SUBST(PANEL_LIBS)
+
     AC_DEFINE(HAVE_APPLETS, 1, [Define if we are building applets])
     AC_MSG_RESULT(yes)
 else
-- 
1.7.0.1

