From 1ec48efc03df9368a59f501dfd1de9c9b15f016c Mon Sep 17 00:00:00 2001
From: Alan Knowles <alan@akbkhome.com>
Date: Fri, 23 Sep 2011 06:08:06 +0000
Subject: fix #659548 - stop CFLAGS getting unconditionally overwritten
Upstream: yes
---
diff --git a/configure.ac b/configure.ac
index adc33b1..b5e6d1b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -392,10 +392,10 @@ AC_ARG_ENABLE(debug,
 
 if test "x$enable_debug" = "xyes"; then
 	SEED_DEBUG_CFLAGS="-DSEED_ENABLE_DEBUG -g"
-	CFLAGS="-O0 -Wall"
+	CFLAGS="$CFLAGS -O0 -Wall"
 else
 	SEED_DEBUG_CFLAGS="-DG_DISABLE_ASSERT -DG_DISABLE_CHECKS -DG_DISABLE_CAST_CHECKS"
-	CFLAGS="-Wall"
+	CFLAGS="$CFLAGS -Wall"
 	
 	if test "x$need_debug" = "xyes"; then
 		AC_MSG_ERROR([debug must be enabled in order to enable profiling.])
--
cgit v0.9.0.2
