From a8dfbb67dc220fa43f7b753ce1f7e0a26f6f48e7 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@exherbo.org>
Date: Thu, 5 Feb 2009 19:52:48 -0800
Subject: [PATCH] Add missing call to AC_PROG_CXX

---
 configure.ac |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index 08d98b6..4d2d961 100644
--- a/configure.ac
+++ b/configure.ac
@@ -68,6 +68,7 @@ WEBKIT_INIT
 AC_DISABLE_STATIC
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
+AC_PROG_CXX
 # use dolt to speedup the build
 DOLT
 
-- 
1.6.1.2

