Author: Saleem Abdulrasool <compnerd@compnerd.org>
Reason: Build without html5 support
Upstream: No
--- 
--- webkit-1.1.5/WebCore/bindings/js/ScriptObjectQuarantine.cpp	2009-04-15 21:12:50.000000000 -0700
+++ webkit-1.1.5/WebCore/bindings/js/ScriptObjectQuarantine.cpp	2009-04-15 21:13:57.000000000 -0700
@@ -31,6 +31,8 @@
 #include "config.h"
 #include "ScriptObjectQuarantine.h"
 
+#if ENABLE(DATABASE)
 #include "Database.h"
+#endif
 #include "Document.h"
 #include "Frame.h"
