--- webkit-1.1.21/WebCore/workers/WorkerThread.cpp.orig	2010-02-09 19:48:27.000000000 -0800
+++ webkit-1.1.21/WebCore/workers/WorkerThread.cpp	2010-02-09 19:49:42.000000000 -0800
@@ -183,6 +183,7 @@
         ASSERT(context->isWorkerContext());
         WorkerContext* workerContext = static_cast<WorkerContext*>(context);
 
+#if ENABLE(DATABASE)
         // We currently ignore any DatabasePolicy used for the document's
         // databases; if it's actually used anywhere, this should be revisited.
         DatabaseTaskSynchronizer cleanupSync;
@@ -194,6 +195,7 @@
         // We wait for the database thread to clean up all its stuff so that we
         // can do more stringent leak checks as we exit.
         cleanupSync.waitForTaskCompletion();
+#endif
 
         // Stick a shutdown command at the end of the queue, so that we deal
         // with all the cleanup tasks the databases post first.
