Author: Saleem Abdulrasool <compnerd@compnerd.org>
Reason: Build without html5 support
Upstream: No
--- 
--- webkit-1.1.5/WebCore/bindings/js/ScriptObjectQuarantine.h	2009-04-15 21:12:52.000000000 -0700
+++ webkit-1.1.5/WebCore/bindings/js/ScriptObjectQuarantine.h	2009-04-15 21:14:45.000000000 -0700
@@ -35,18 +35,26 @@
 
 namespace WebCore {
 
+#if ENABLE(DATABASE)
     class Database;
+#endif
     class DOMWindow;
     class Frame;
     class Node;
     class ScriptObject;
     class ScriptValue;
+#if ENABLE(STORAGE)
     class Storage;
+#endif
 
     ScriptValue quarantineValue(ScriptState*, const ScriptValue&);
 
+#if ENABLE(DATABASE)
     bool getQuarantinedScriptObject(Database* database, ScriptObject& quarantinedObject);
+#endif
+#if ENABLE(STORAGE)
     bool getQuarantinedScriptObject(Frame* frame, Storage* storage, ScriptObject& quarantinedObject);
+#endif
     bool getQuarantinedScriptObject(Node* node, ScriptObject& quarantinedObject);
     bool getQuarantinedScriptObject(DOMWindow* domWindow, ScriptObject& quarantinedObject);
 
--- webkit-1.1.5/WebCore/bindings/js/ScriptObjectQuarantine.cpp	2009-04-15 21:12:50.000000000 -0700
+++ webkit-1.1.5/WebCore/bindings/js/ScriptObjectQuarantine.cpp	2009-04-15 21:13:57.000000000 -0700
@@ -31,11 +31,17 @@
 #include "config.h"
 #include "ScriptObjectQuarantine.h"
 
+#if ENABLE(DATABASE)
 #include "Database.h"
+#endif
 #include "Document.h"
 #include "Frame.h"
+#if ENABLE(DATABASE)
 #include "JSDatabase.h"
+#endif
+#if ENABLE(STORAGE)
 #include "JSStorage.h"
+#endif
 #include "JSDOMBinding.h"
 #include "JSInspectedObjectWrapper.h"
 #include "JSNode.h"
@@ -54,6 +60,7 @@
     return ScriptValue(JSInspectedObjectWrapper::wrap(scriptState, value.jsValue()));
 }
 
+#if ENABLE(DATABASE)
 bool getQuarantinedScriptObject(Database* database, ScriptObject& quarantinedObject)
 {
     ASSERT(database);
@@ -69,7 +76,9 @@
 
     return true;
 }
+#endif
 
+#if ENABLE(STORAGE)
 bool getQuarantinedScriptObject(Frame* frame, Storage* storage, ScriptObject& quarantinedObject)
 {
     ASSERT(frame);
@@ -82,6 +91,7 @@
 
     return true;
 }
+#endif
 
 bool getQuarantinedScriptObject(Node* node, ScriptObject& quarantinedObject)
 {
