--- webkit-1.1.13/configure.ac	2009-08-25 20:59:56.000000000 -0700
+++ webkit-1.1.13/configure.ac	2009-08-25 21:00:23.000000000 -0700
@@ -81,6 +81,7 @@
 WEBKIT_INIT
 AC_DISABLE_STATIC
 AC_LIBTOOL_WIN32_DLL
+AC_PROG_CXX
 AC_PROG_LIBTOOL
 # use dolt to speedup the build
 DOLT
--- webkit-1.1.13/GNUmakefile.am	2009-08-25 20:50:37.000000000 -0700
+++ webkit-1.1.13/GNUmakefile.am	2009-08-25 20:52:01.000000000 -0700
@@ -351,7 +351,6 @@
 	WebKit/gtk/WebCoreSupport/InspectorClientGtk.h \
 	WebKit/gtk/WebCoreSupport/PasteboardHelperGtk.cpp \
 	WebKit/gtk/WebCoreSupport/PasteboardHelperGtk.h \
-	WebKit/gtk/webkit/webkitapplicationcache.cpp \
 	WebKit/gtk/webkit/webkitdownload.cpp \
 	WebKit/gtk/webkit/webkiterror.cpp \
 	WebKit/gtk/webkit/webkitnetworkrequest.cpp \
@@ -370,6 +369,11 @@
 	WebKit/gtk/webkit/webkitwebwindowfeatures.cpp \
 	WebKit/gtk/webkit/webkitworkers.cpp
 
+if ENABLE_DOM_STORAGE
+webkitgtk_sources += \
+	WebKit/gtk/webkit/webkitapplicationcache.cpp
+endif
+
 webkitgtk_cppflags += \
 	-DBUILDING_WEBKIT \
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\" \
--- webkit-1.1.13/WebCore/bindings/js/ScriptObjectQuarantine.cpp	2009-08-25 21:25:46.000000000 -0700
+++ webkit-1.1.13/WebCore/bindings/js/ScriptObjectQuarantine.cpp	2009-08-25 21:27:42.000000000 -0700
@@ -38,7 +38,9 @@
 #include "JSNode.h"
 #include "ScriptObject.h"
 #include "ScriptValue.h"
+#if ENABLE(DOM_STORAGE)
 #include "Storage.h"
+#endif
 
 #include <runtime/JSLock.h>
 
--- webkit-1.1.13/WebCore/page/Navigator.cpp	2009-08-25 21:36:52.000000000 -0700
+++ webkit-1.1.13/WebCore/page/Navigator.cpp	2009-08-25 21:37:16.000000000 -0700
@@ -37,7 +37,9 @@
 #include "PluginData.h"
 #include "ScriptController.h"
 #include "Settings.h"
+#if ENABLE(DOM_STORAGE)
 #include "StorageNamespace.h"
+#endif
 
 namespace WebCore {
 
--- webkit-1.1.13/WebKitTools/DumpRenderTree/gtk/LayoutTestControllerGtk.cpp	2009-08-25 21:50:45.000000000 -0700
+++ webkit-1.1.13/WebKitTools/DumpRenderTree/gtk/LayoutTestControllerGtk.cpp	2009-08-25 21:52:19.000000000 -0700
@@ -354,10 +354,12 @@
     // FIXME: implement
 }
 
+#if ENABLE(DOM_STORAGE)
 void LayoutTestController::setAppCacheMaximumSize(unsigned long long size)
 {
     webkit_application_cache_set_maximum_size(size);
 }
+#endif
 
 bool LayoutTestController::pauseAnimationAtTimeOnElementWithId(JSStringRef animationName, double time, JSStringRef elementId)
 {    
--- webkit-1.1.13/WebKitTools/DumpRenderTree/LayoutTestController.cpp	2009-08-25 21:56:20.000000000 -0700
+++ webkit-1.1.13/WebKitTools/DumpRenderTree/LayoutTestController.cpp	2009-08-25 21:57:42.000000000 -0700
@@ -535,6 +535,7 @@
     return JSValueMakeUndefined(context);
 }
 
+#if ENABLE(DOM_STORAGE)
 static JSValueRef setAppCacheMaximumSizeCallback(JSContextRef context, JSObjectRef function, JSObjectRef thisObject, size_t argumentCount, const JSValueRef arguments[], JSValueRef* exception)
 {
     // Has mac implementation
@@ -550,6 +551,7 @@
     return JSValueMakeUndefined(context);
 
 }
+#endif
 
 static JSValueRef setAuthorAndUserStylesEnabledCallback(JSContextRef context, JSObjectRef function, JSObjectRef thisObject, size_t argumentCount, const JSValueRef arguments[], JSValueRef* exception)
 {
--- webkit-1.1.13/WebKitTools/DumpRenderTree/LayoutTestController.cpp	2009-08-25 21:59:11.000000000 -0700
+++ webkit-1.1.13/WebKitTools/DumpRenderTree/LayoutTestController.cpp	2009-08-25 21:59:55.000000000 -0700
@@ -1013,7 +1013,9 @@
         { "repaintSweepHorizontally", repaintSweepHorizontallyCallback, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete },
         { "setAcceptsEditing", setAcceptsEditingCallback, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete },
         { "setAuthorAndUserStylesEnabled", setAuthorAndUserStylesEnabledCallback, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete },
+#if ENABLE(DOM_STORAGE)
         { "setAppCacheMaximumSize", setAppCacheMaximumSizeCallback, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete }, 
+#endif
         { "setCallCloseOnWebViews", setCallCloseOnWebViewsCallback, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete },
         { "setCanOpenWindows", setCanOpenWindowsCallback, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete },
         { "setCacheModel", setCacheModelCallback, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete },
--- webkit-1.1.14/WebKit/gtk/webkit/webkitwebdatabase.cpp.orig	2009-09-09 21:33:34.000000000 -0700
+++ webkit-1.1.14/WebKit/gtk/webkit/webkitwebdatabase.cpp	2009-09-09 21:34:15.000000000 -0700
@@ -23,8 +23,10 @@
 #include "webkitprivate.h"
 
 #include "CString.h"
+#if ENABLE(DOM_STORAGE)
 #include "DatabaseDetails.h"
 #include "DatabaseTracker.h"
+#endif
 
 #include <glib/gi18n-lib.h>
 
--- webkit-1.1.14/WebKit/gtk/webkit/webkitsecurityorigin.cpp.orig	2009-09-09 21:34:44.000000000 -0700
+++ webkit-1.1.14/WebKit/gtk/webkit/webkitsecurityorigin.cpp	2009-09-09 21:34:56.000000000 -0700
@@ -24,7 +24,9 @@
 
 #include "CString.h"
 #include "PlatformString.h"
+#if ENABLE(DOM_STORAGE)
 #include "DatabaseTracker.h"
+#endif
 
 #include <glib/gi18n-lib.h>
 
--- webkit-1.1.14/WebCore/bindings/js/ScriptController.cpp.orig	2009-09-09 21:43:42.000000000 -0700
+++ webkit-1.1.14/WebCore/bindings/js/ScriptController.cpp	2009-09-09 21:43:58.000000000 -0700
@@ -33,7 +33,9 @@
 #include "ScriptSourceCode.h"
 #include "ScriptValue.h"
 #include "Settings.h"
+#if ENABLE(DOM_STORAGE)
 #include "StorageNamespace.h"
+#endif
 #include "XSSAuditor.h"
 #include "npruntime_impl.h"
 #include "runtime_root.h"
