From 6ae79e55bfb57ace67168ef031760bd997832d35 Mon Sep 17 00:00:00 2001
From: Saleem Abdulrasool <compnerd@compnerd.org>
Date: Thu, 10 Mar 2011 22:40:02 -0800
Subject: [PATCH 2/2] shell: fix toolbar style enum

---
 lib/ephy-prefs.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/ephy-prefs.h b/lib/ephy-prefs.h
index d4a9755..45ca6ba 100644
--- a/lib/ephy-prefs.h
+++ b/lib/ephy-prefs.h
@@ -31,8 +31,8 @@ typedef enum
 {
 	EPHY_PREFS_UI_TOOLBAR_STYLE_BOTH,
 	EPHY_PREFS_UI_TOOLBAR_STYLE_BOTH_HORIZ,
-	EPHY_PREFS_UI_TOOLBAR_STYLE_BOTH_ICONS,
-	EPHY_PREFS_UI_TOOLBAR_STYLE_BOTH_TEXT
+	EPHY_PREFS_UI_TOOLBAR_STYLE_ICONS,
+	EPHY_PREFS_UI_TOOLBAR_STYLE_TEXT
 } EphyPrefsUIToolbarStyle;
 
 typedef enum
-- 
1.7.4.1

